Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/IdentifierProvider.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.dataformat.access;
+
+import java.util.Iterator;
+
+/**
+ * Provides unique identifiers for elements in XML files.
+ */
+class IdentifierProvider implements Iterator<String> {
```

Comment:
Why are you not using UUID?
For whatever reason you don't, then why are you implementing Iterator<String>?
Everything you need is:

```
private static long idCounter = 0;

public static synchronized String createID()
{
    return String.valueOf(idCounter++);
} 
```

Reply:
I chose integer, because they are pretty short. If you want to use UUID we can still replace it—provided it works. I do not know what a UUID must contain and I do not know if METS IDs are allowed to contain all characters. I have implemented iterator because I find it so clear. But we can also leave it out. Should I remove it?

Reply:
`static synchronized` is also nonsense. I do not want to allocate identifiers across Production uniquely (and then reset the counter back to 1 at each server restart). The identifiers need only be unique within a METS file so that the XML parser knows what is referring to what:

![id_referencing](https://user-images.githubusercontent.com/3040657/49806412-035a5080-fd58-11e8-8bd2-801aedde0e37.png)